axelrod/fingerprint.py

```diff
@@ -0,0 +1,226 @@
+        step : float
+            The separation between each coordinate. Smaller steps will
+            produce more coordinates that will be closer together.
+
+        Returns
+        ----------
+        coordinates : list of tuples
+            Tuples of length 2 representing each coordinate, eg. (x, y)
+        """
+        coordinates = list(product(np.arange(0, 1, step),
+                                   np.arange(0, 1, step)))
+        return coordinates
+
+    def create_probes(self, probe, probe_coords):
+        """Creates a set of probe strategies over the unit square.
+
+        Constructs probe strategies that correspond to (x, y) coordinates. The
+        precision of the coordinates is determined by `step`. The probes
+        are created using the `JossAnnTransformer`.
+
+        Parameters
+        ----------
+        probe : class
+            A class that must be descended from axelrod.strategies.
+        probe_coords : ordered dictionary
+            An Ordered Dictionary where the keys are tuples representing each
+            coordinate, eg. (x, y). The value is automatically set to `None`.
+
+        Returns
+        ----------
+        probe_dict : ordered dictionary
+            An Ordered Dictionary where the keys are tuples representing each
+            coordinate, eg. (x, y). The value is a `JossAnnTransformer` with
+            parameters that correspond to (x, y).
+        """
+        probe_dict = OrderedDict((coord, create_jossann(coord, probe))
+                                 for coord in probe_coords)
+        return probe_dict
+
+    def create_edges(self, probe_dict):
+        """Creates a set of edges for a spatial tournament.
+
+        Constructs edges that correspond to the probes in `probe_dict`. Probes
+        whose coordinates sum to less/more than 1 will have edges that link them
+        to 0/1 correspondingly.
+
+        Parameters
+        ----------
+        probe_dict : ordered dictionary
+            An Ordered Dictionary where the keys are tuples representing each
+            coordinate, eg. (x, y). The value is a `JossAnnTransformer` with
+            parameters that correspond to (x, y).
+
+        Returns
+        ----------
+        edges : list of tuples
+            A list containing tuples of length 2. All tuples will have either 0
+            or 1 as the first element. The second element is the index of the
+            corresponding probe (+2 to allow for including the Strategy and it's
+            Dual).
+        """
+        edges = []
+        for index, coord in enumerate(probe_dict.keys()):
+            #  Add 2 to the index because we will have to allow for the Strategy
+            #  and it's Dual
+            if sum(coord) > 1:
+                edge = (1, index + 2)
+            else:
+                edge = (0, index + 2)
+            edges.append(edge)
+        return edges
+
+    def fingerprint(self, turns=50, repetitions=10, step=0.01, processes=None):
```

Comment:
This method has a lot of side-effects but returns nothing. That makes it tricky to test and even more difficult to debug when things don't work as expected.

I think it needs refactoring into one or more functions/methods with defined return values, tests, and no side-effectd.